CONTRIBUTING.md

```diff
@@ -29,26 +31,119 @@ The [PyScript Code of Conduct](https://github.com/pyscript/governance/blob/main/
+
+### PR approval requirements
+
+After you've created a PR there are a few extra steps that need to be taken so the PR can be approved. These steps aim to make sure the approval process is clear and that when a PR is merged, the level of quality of the codebase meets the standards set by the project `maintainers` in order to avoid regression and, ultimately, to keep high standards for the project users.
+
+If your PR changes or adds functionality, it's expected that it includes:
+
+**Description**: Every PR should have a meaningful and exhaustive description of what they do and a link to the original issue they address.
+
+**Title**: A descriptive title for your PR will help others find and refer to it consistently.
+
+**Tests**: This is the way we ensure added/modified functionality meets the related feature expectations and that other code changes don't break functionlity over time.
+
+**Documentation**: If your PR changes or adds functionality, it's expected that you also make sure to add the right information to the docs. For more information about how to work on the project documentation check out the [Setting Up Your Documentation Environment](https://github.com/pyscript/pyscript/blob/main/docs/development/setting-up-environment.md#setting-up-your-documentation-environment).
+
+Additionally, PR's which change user-facing functionality are expected to include an entry in the Changelog, for users' quick reference when comparing versions of PyScript. For information on updating the changelog, see [Updating the Changelog](https://github.com/pyscript/pyscript/blob/main/docs/development/developing.md#updating-the-changelog).
+
+**Readable code**: Ok, this is a broad and generic statement and deserves some explanation. The mantra behind this statement is that code that is semantically meaningful (aka "has a meaning" :-) is preferable to code that is short and "saves some line of code". Remember, others will be reading your code and if they understand what's going on easily it's a good sign that your code reflects your intentions while also making it easier for everyone. In addition to that, remember that you may be reading that code in the future and your future self will definitely thank you for making it easy to read! For instance, a code like:
+
+```python
+t = "python"
+...
+# 20 lines later...
+xs = engine.fetch_plugins(t)
+
+...
+# another 20 lines later...
+for x in xs:
+    x.init()
+```
+
+is harder to read compare to something like:
+
+```python
+plugin_type = "python"
+...
+# 20 lines later...
+plugins = engine.fetch_plugins(plugin_type)
+
+...
+# another 20 lines later...
+for plugin in plugins:
+    plugin.init()
+```
+
+**Docstrings**: Document. Your. Code! Make sure all your classes and functions have docstrings that document their behaviour, inputs and outputs. Other people (especially users!) shouldn't need to dive your the code implementing a function in order to understand what it does.
```

Comment:
nit: I don't agree with this but it surely helps debugging issues ... it's just weird we have the policy of "*one char reference being good enough*" in our linter rules but we're asking for extra verbosity out there.

Reply:
to clarify, tools have zero issues expanding on terser references names but we really have a lot of code preferring `r` instead of `runtime` (as made up example) so I'm not sure this helps anyone concretely out there, when we're relaxed ourselves about this rule ... dare I say: make it pass our linter or try harder would be my preferred option.